### PR TITLE
Scripts for patch releases

### PR DESCRIPTION
**Pull request by msau42**
Added a script to update specific go modules, and fixed the release notes script to not depend on discover mode which seems to be broken.

### CODE REVIEW

File: generate-patch-release-notes.sh [hunk 1]
```diff
@@ -48,7 +48,7 @@ function gen_patch_relnotes() {
   rm out.md || true
```
Comment (carlory) [inline generate-patch-release-notes.sh:48]: after I run this script, it doesn't cleanup the out.md.

```
(⎈|kind-kind:N/A)➜  csi-driver-host-path git:(changelog-release-1.12) ✗ git status
On branch changelog-release-1.12
Your branch is ahead of 'upstream/release-1.12' by 1 commit.
  (use "git push" to publish your local commits)

Untracked files:
  (use "git add <file>..." to include in what will be committed)
	CHANGELOG/out.md

nothing added to commit but untracked files present (use "git add" to track)
```

Comment (msau42) [reply generate-patch-release-notes.sh:48]: fixed

**Comment by carlory** [comment]
/remove-lifecycle stale
/lgtm
/hold 

feel free to cancel the hold

**Pull request merged**